docs/rfcs/20220131_raft_snapshots.md

```diff
@@ -0,0 +1,149 @@
+- Feature Name: Raft snapshots
+- Status: draft
+- Start Date: 2022-01-31
+- Authors: Denis Rystsov
+- Issue: TBD
+
+# Executive Summary
+
+Introduce Raft snapshots not bound to log truncation to solve compaction related problems with idempotency, transactions, offset translator and pave a way to custom state machines.
+
+## What is being proposed
+
```

Comment:
Question, when compaction is performed, why can't replicas just apply the compaction themselves? Why shouldn't the same results be observed everywhere?

Reply:
Not really, a follower may be isolated from the cluster for a while and all the other nodes may already did the compaction so that when the node rejoins the cluster it already pulls the log with the holes. Another case when it happens is the reconfiguration when we move the partition from nodes `a,b,c` to nodes `d,e,f` in this case the newer nodes never saw the original data, only already compacted or truncated.